# This patch makes sure we have access to uprv_getICUData_conversion

diff --git a/icu4c/source/common/udata.cpp b/icu4c/source/common/udata.cpp
index ec9c999cea..2e6ac0ea3b 100644
--- source/common/udata.cpp
+++ source/common/udata.cpp
@@ -18,11 +18,10 @@

 #include "unicode/utypes.h"  /* U_PLATFORM etc. */

-#ifdef __GNUC__
-/* if gcc
-#define ATTRIBUTE_WEAK __attribute__ ((weak))
-might have to #include some other header
-*/
+#if defined(__GNUC__) || defined(__SUNPRO_CC)
+#  define ATTRIBUTE_WEAK __attribute__ ((weak))
+#else
+#  define ATTRIBUTE_WEAK
 #endif

 #include "unicode/putil.h"
@@ -651,10 +650,9 @@ extern "C" const DataHeader U_DATA_API U_ICUDATA_ENTRY_POINT;
  * partial-data-library access functions where each returns a pointer
  * to its data package, if it is linked in.
  */
-/*
-extern const void *uprv_getICUData_collation(void) ATTRIBUTE_WEAK;
+U_CDECL_BEGIN
 extern const void *uprv_getICUData_conversion(void) ATTRIBUTE_WEAK;
-*/
+U_CDECL_END

 /*----------------------------------------------------------------------*
  *                                                                      *
@@ -713,10 +711,10 @@ openCommonData(const char *path,          /*  Path from OpenChoice?          */
         if (uprv_getICUData_collation) {
             setCommonICUDataPointer(uprv_getICUData_collation(), FALSE, pErrorCode);
         }
+	*/
         if (uprv_getICUData_conversion) {
             setCommonICUDataPointer(uprv_getICUData_conversion(), FALSE, pErrorCode);
         }
-        */
 #if !defined(ICU_DATA_DIR_WINDOWS)
 // When using the Windows system data, we expect only a single data file.
         setCommonICUDataPointer(&U_ICUDATA_ENTRY_POINT, FALSE, pErrorCode);
diff --git a/icu4c/source/common/unicode/uconfig.h b/icu4c/source/common/unicode/uconfig.h
index bbc232d1ed..466bb68c92 100644
--- source/common/unicode/uconfig.h
+++ source/common/unicode/uconfig.h
@@ -55,6 +55,10 @@
 #include "uconfig_local.h"
 #endif

+#ifndef U_STATIC_IMPLEMENTATION
+#define U_STATIC_IMPLEMENTATION
+#endif
+
 /**
  * \def U_DEBUG
  * Determines whether to include debugging code.
